### PR TITLE
HPCC-13245 Improve temporary row code for no_selectnth

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jamienoss another one to review

File: ecl/hqlcpp/hqlcppds.cpp [hunk 1]
```diff
@@ -4373,6 +4373,14 @@ void HqlCppTranslator::buildRowAssign(BuildCtx & ctx, BoundRow * targetRow, IHql
             buildStmt(ctx, expr->queryChild(0));
             buildRowAssign(ctx, targetRow, expr->queryChild(1));
             return;
+        case no_selectnth:
```
Comment (jamienoss) [inline ecl/hqlcpp/hqlcppds.cpp]: How much of the comment starting on line 4353 is still valid?

Comment (jamienoss) [inline ecl/hqlcpp/hqlcppds.cpp]: Is no_selectnth the only case that can allow for row linking?

**Comment by jamienoss** [comment]
@ghalliday this looks good though please see my only comment.
In addition, perhaps some amendment to the jira summery/description is required to account for this only applying to no_selectnth.

File: ecl/hqlcpp/hqlcppds.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hqlcpp/hqlcppds.cpp]: Still generally valid.  I opened https://track.hpccsystems.com/browse/HPCC-13252 to cover improving it.

**Pull request merged**